pandas/core/arrays/boolean.py

```diff
@@ -696,9 +696,18 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
         data = self._data
         mask = self._mask
 
-        if name in {"sum", "min", "max"}:
+        if name in {"sum", "prod", "min", "max"}:
             op = getattr(masked_reductions, name)
-            return op(data, mask, skipna=skipna, **kwargs)
+            result = op(data, mask, skipna=skipna, **kwargs)
+
```

Comment:
can you try using maybe_cast_result_dtype here (you will have to add the prod operation in side that)

Reply:
As far as I understand, that should not be needed. Numpy should give the desired result for booleans.